lib/libxdp/xsk.c

```diff
@@ -683,6 +713,60 @@ static struct xdp_program *xsk_lookup_program(int ifindex)
 	return prog;
 }
 
+static int xsk_update_prog_refcnt(int refcnt_map_fd, int delta)
+{
+	struct bpf_map_info map_info = {};
+	__u32 info_len = sizeof(map_info);
+	__u32 refcnt_offset = 0, key = 0;
+	int lock_fd, ret, *value;
+	void *value_data = NULL;
+
+	lock_fd = xdp_lock_acquire();
```

Comment:
Nit: You could move the `lock_acquire()` below the malloc call, as neither `get_info_by_fd()` nor `malloc()` strictly speaking needs to be protected by the lock; this will hold the lock for a slightly shorter time, and will save the check on value_data in the out path below...